nbformat/validator.py

```diff
@@ -270,15 +271,17 @@ def validate(nbdict=None, ref=None, version=None, version_minor=None,
                                relax_add_props=relax_add_props):
         raise error
 
-    if ref is None and version >= 4 and version_minor >= 5:
+    if notebook_supports_cell_ids:
         # if we support cell ids check for uniqueness when validating the whole notebook
         seen_ids = set()
         for cell in nbdict['cells']:
             cell_id = cell['id']
             if cell_id in seen_ids:
-                cell['id'] = generate_corpus_id()
-                # Best effort to repair if we find a duplicate id in case the ValidationError isn't blocking
-                raise ValidationError("Non-unique cell id '{}' detected. Corrected to '{}'.".format(cell_id, cell['id']))
+                if repair:
+                    # Best effort to repair if we find a duplicate id
+                    cell['id'] = generate_corpus_id()
```

Comment:
Since normally we log by catching that validation message, let's log a warning indicating that we repaired a conflicting id